workspaces/diff-engine/src/streams/spec_chunks.rs

```diff
@@ -0,0 +1,89 @@
+use super::spec_events;
+use crate::{
+  events::{spec_chunk, SpecChunkEvent},
+  SpecAssemblerProjection, SpecEvent,
+};
+use fs::{read_dir, read_to_string};
+use std::ffi::OsString;
+use std::path::Path;
+use tokio::{fs, io};
+use tokio_stream::wrappers::ReadDirStream;
+use tokio_stream::{StreamExt, StreamMap};
+
+// TODO: return a stream instead of a Vec
+pub async fn from_api_dir(
+  path: impl AsRef<Path>,
+) -> Result<Vec<SpecChunkEvent>, SpecChunkLoaderError> {
+  let read_dir = fs::read_dir(path).await?;
+  let mut dir_entries = ReadDirStream::new(read_dir);
+
+  let mut chunks = Vec::<SpecChunkEvent>::new();
+  while let Some(dir_entry_result) = dir_entries.next().await {
+    let dir_entry = dir_entry_result?;
+
+    let metadata = dir_entry.metadata().await?;
+
+    if metadata.is_file() {
+      let file_name = dir_entry.file_name();
+      let file_path = dir_entry.path();
+
+      let spec_events_result = spec_events::from_file(&file_path).await;
+      if spec_events_result.is_err() {
+        // TODO: consider logging this better
+        eprintln!(
+          "skipping file: not valid spec events path={:?}: {:?}",
+          &file_path,
+          spec_events_result.unwrap_err()
+        );
+        continue;
+      }
+      let spec_events = spec_events_result.unwrap();
+      let name = file_name.into_string().map_err(|_| {
+        SpecChunkLoaderError::Other("Filename could not be converted to valid UTF-8")
+      })?;
+      let is_root = name == "specification.json";
+
+      chunks.push(SpecChunkEvent::from((name, is_root, spec_events)));
+    }
+  }
+
+  Ok(chunks)
+}
```

Comment:
The SpecChunk stream is where all the logic lives around reading spec-change files from a directory and mapping them to `SpecChunkEvent` instances. Those paying attention will notice how this doesn't return a `impl Stream<Item=SpecChunkEvent>`, but that actually turned out be just tricky enough for me to decide it's something to consider in the future instead of now. 

However, it _does_ still belong in this module, as all of `streams` is not imported for `diff-engine-wasm`, which is important as Tokio + wasm isn't a love story yet. Reading files async from the file system this way is great, but requires that Tokio runtime, so best isolated away from the rest of the code.